### PR TITLE
e2e test for domain verification

### LINKED ISSUE #344: e2e test for custom domains
Priority: Medium

## Issue Description
Create an e2e test for custom domains

### PR DESCRIPTION
**Pull request by philbrookes**
Closes #344

## Description of Changes
- add e2e for domain verification
- add fake DNS resolver for e2e tests
- add e2e helper functions for use with fake DNS Resolver
- add e2e helper functions for configmaps
- add e2e helper functions for Ingress domain verification
- change finalizer from kcp.dev to kuadrant.dev

## Release and Testing
See the e2e tests have passed

### CODE REVIEW

File: utils/local-setup.sh [hunk 1]
```diff
@@ -71,7 +71,7 @@ ORG_WORKSPACE=root:default
 KUBECONFIG_KCP_ADMIN=.kcp/admin.kubeconfig
 KUBECONFIG_KCP_GLBC=${TEMP_DIR}/kcp.kubeconfig
 
-: ${KCP_VERSION:="release-0.7"}
+: ${KCP_VERSION:="release-0.8"}
```
Comment (mikenairn) [inline utils/local-setup.sh:74]: This is only one of a few things that need updated when updating KPC versions (Your e2e tests are still using 0.7). There are also other known changes that will need to be made when moving from 0.7 -> 0.8. I think you should remove this and we can do the KCP 0.8 update in it's own PR. Added a separate issue for it https://github.com/Kuadrant/kcp-glbc/issues/362

Comment (philbrookes) [reply utils/local-setup.sh]: ok, done.

File: config/rbac/role.yaml [hunk 2]
```diff
@@ -7,6 +7,7 @@ rules:
       - ""
     resources:
       - secrets
+      - configmaps
```
Comment (maleck13) [inline config/rbac/role.yaml:10]: hmm not sure about this change? We don't actually need that access except in the test right?

File: cmd/kcp-glbc/main.go [hunk 3]
```diff
@@ -101,6 +104,8 @@ func init() {
 	flagSet.StringVar(&options.Domain, "domain", env.GetEnvString("GLBC_DOMAIN", "dev.hcpapps.net"), "The domain to use to expose ingresses")
 	flagSet.BoolVar(&options.EnableCustomHosts, "enable-custom-hosts", env.GetEnvBool("GLBC_ENABLE_CUSTOM_HOSTS", false), "Flag to enable hosts to be custom")
 	flag.StringVar(&options.DNSProvider, "dns-provider", env.GetEnvString("GLBC_DNS_PROVIDER", "fake"), "The DNS provider being used [aws, fake]")
+	flagSet.StringVar(&options.HostResolver, "host-resolver", env.GetEnvString("GLBC_HOST_RESOLVER", "default"), "The Host resolver to use, one of [default, e2e-mock]")
```
Comment (mikenairn) [inline cmd/kcp-glbc/main.go:107]: Is changing this something we'd ever do apart from mocking in the e2e test?

File: config/deploy/local/controller-config.env.ci [hunk 4]
```diff
@@ -2,11 +2,13 @@ AWS_DNS_PUBLIC_ZONE_ID=FAKE_ZONE_ID
...
 GLBC_DOMAIN=dev.hcpapps.net
-GLBC_ENABLE_CUSTOM_HOSTS=false
+GLBC_ENABLE_CUSTOM_HOSTS=true
 GLBC_EXPORT=glbc-root-default-kcp-glbc-user
+GLBC_HOST_RESOLVER=e2e-mock
 GLBC_LOGICAL_CLUSTER_TARGET=*
 GLBC_TLS_PROVIDED=true
 GLBC_TLS_PROVIDER=glbc-ca
 GLBC_WORKSPACE=root:default:kcp-glbc
+GLBC_ADVANCED_SCHEDULING=true
```
[the remark this answers is not in the document]
Comment (philbrookes) [reply config/deploy/local/controller-config.env.ci:12]: agreed, I think something centralised to enable or disable this for sync targets and GLBC at the same time is a good idea for the future.

File: config/rbac/role.yaml (hunk 2 above)
Comment (philbrookes) [reply config/rbac/role.yaml:10]: Actually, I don't think this is required anyway. I will remove it

File: cmd/kcp-glbc/main.go (hunk 3 above)
Comment (philbrookes) [reply cmd/kcp-glbc/main.go:107]: No, it isn't. Do you think I should just have it as an envvar and not a CLI option, then? Or is there a better way to pass in configuration that is just for e2e tests?

Comment (philbrookes) [reply cmd/kcp-glbc/main.go:107]: moved to just an envvar

File: pkg/reconciler/ingress/status.go [hunk 5]
```diff
@@ -19,7 +20,7 @@ func GetStatus(i *networkingv1.Ingress) (map[logicalcluster.Name]*networkingv1.I
 			continue
 		}
 		annotationParts := strings.Split(k, "/")
-		fmt.Println("get status annoation parts", annotationParts)
+		fmt.Println("get status annotation parts", annotationParts)
```
Comment (maleck13) [inline pkg/reconciler/ingress/status.go:23]: I think we can remove this it is a stray log

**Review by maleck13 — APPROVED** [review]
Would like if we removed the fmt.Println but changes look good and I know that was tehre before

File: pkg/reconciler/ingress/status.go (hunk 5 above)
Comment (philbrookes) [reply pkg/reconciler/ingress/status.go]: done

**Pull request merged**